src/libraries/System.IO/tests/StreamReader/StreamReader.cs

```diff
@@ -72,5 +72,30 @@ public void DetectEncoding_EncodingRoundtrips(Encoding encoding)
                 Assert.Equal(encoding.EncodingName, reader.CurrentEncoding.EncodingName);
             }
         }
+
+        [Theory]
+        [MemberData(nameof(DetectEncoding_EncodingRoundtrips_MemberData))]
+        public void DetectEncoding_EncodingRoundtrips_Path(Encoding encoding)
+        {
+            const string Text = "This is some text for testing.";
+            string path = GetTestFilePath();
+
+            using (var writer = new StreamWriter(path, false, encoding))
```

Comment:
I would expect to see `FileStreamOptions` being used here (as this is the ctor you are adding in this PR)